tests/lib/rules/no-unguarded-globals.js

```diff
@@ -504,6 +504,72 @@ ruleTester.run('no-unguarded-globals', rule, {
       errors: [{
         message: DOCUMENT_MESSAGE
       }]
+    },
+    {
+      code: `
+        import environment from 'ember-stdlib/utils/environment';
+
+        export default Ember.Component.extend({
+          shortCircuit() {
+            if (environment.isBrowser() && window.localStorage) {
+              return window.localStorage;
+            }
+          }
+        });`,
+      parserOptions: {
+        ecmaVersion: 6,
+        sourceType: 'module'
+      }
+    },
+    {
+      code: `
+        import environment from 'ember-stdlib/utils/environment';
+        const { isBrowser } = environment
+
+        export default Ember.Component.extend({
+          shortCircuit() {
+            if (isBrowser() && window.localStorage) {
+              return window.localStorage;
+            }
+          }
+        });`,
+      parserOptions: {
+        ecmaVersion: 6,
+        sourceType: 'module'
+      }
+    },
+    {
+      code: `
+        import environment from 'ember-stdlib/utils/environment';
+
+        export default Ember.Component.extend({
+          isInBrowser() {
+            if (environment.isBrowser() || window) {
```

Comment:
Will this work if one does:

```js
return environment.isBrowser() || window;
```
?

Reply:
That is marked as invalid. Will add a test for it

Reply:
Great, thanks!